bin/install.sh

```diff
@@ -30,80 +30,116 @@ function install_crds() {
 # Install citadel into namespace istio-system
 
 function install_system() {
-    kubectl delete namespace istio-system --wait --ignore-not-found
-    echo "Installing citadel.."
-    bin/iop istio-system istio-system-security $IBASE/security/citadel/ --set global.defaultResources.requests.cpu=0m
+    if [ "$METHOD" = Install ]; then
+        kubectl delete namespace istio-system --wait --ignore-not-found
+    fi
+    echo "${METHOD}ing citadel.."
+    bin/iop istio-system istio-system-security $IBASE/security/citadel/ $RESOURCES_FLAGS
     
-    kubectl get deployments -n istio-system
-    kubectl wait deployments istio-citadel11 -n istio-system --for=condition=available --timeout=$WAIT_TIMEOUT
     kubectl rollout status  deployment istio-citadel11 -n istio-system --timeout=$WAIT_TIMEOUT
-    kubectl get deployments -n istio-system
-    kubectl get pod -n istio-system
 }
 
 # Install config, discovery and sidecar-injector into namespace istio-control
 
 function install_control() {
-    kubectl delete namespace istio-control --wait --ignore-not-found
-    echo "Installing galley.."
-    bin/iop istio-control istio-config $IBASE/istio-control/istio-config --set configValidation=true $RESOURCES_FLAGS
-    echo "Installing pilot.."
-    bin/iop istio-control istio-discovery $IBASE/istio-control/istio-discovery $RESOURCES_FLAGS
-    echo "Installing auto-injector.."
-    bin/iop istio-control istio-autoinject $IBASE/istio-control/istio-autoinject --set global.istioNamespace=istio-control $RESOURCES_FLAGS
-
-    kubectl get deployments -n istio-control
-    kubectl wait deployments istio-galley istio-pilot istio-sidecar-injector -n istio-control --for=condition=available --timeout=$WAIT_TIMEOUT
-    kubectl rollout status  deployment istio-galley -n istio-control --timeout=$WAIT_TIMEOUT 
-    kubectl rollout status  deployment istio-pilot  -n istio-control --timeout=$WAIT_TIMEOUT 
-    kubectl rollout status  deployment istio-sidecar-injector -n istio-control --timeout=$WAIT_TIMEOUT
-
-    kubectl get deployments -n istio-control
-    kubectl get pod -n istio-control 
+    kubectl delete namespace ${ISTIO_CONTROL_NS} --wait --ignore-not-found
```

Comment:
Can you skip delete ? 

In all CI/CD systems the test is run in a clean cluster AFAIK.